packages/@react-spectrum/menu/src/ActionMenu.tsx

```diff
@@ -40,7 +41,8 @@ function ActionMenu<T extends object>(props: SpectrumActionMenuProps<T>, ref: Fo
       <ActionButton
         ref={ref}
         {...buttonProps}
-        UNSAFE_className={props.UNSAFE_className}
+        UNSAFE_className={styleProps.className}
+        UNSAFE_style={styleProps.style}
```

Comment:
Why not pass through other props from `props` here? This will miss some style props (e.g. `hidden`), and more if we change the way they work in the future.

Reply:
good call, forgot about hidden, would we do it like
`{...styleProps as Omit<StyleProps, 'className' | 'style'>}`
or would a utility to re-convert to UNSAFE be a good idea for cases like this?

Reply:
why does it need to be unsafe? You don't need to call `useStyleProps` at all here, just pass them through and let ActionButton handle it?

```js
<ActionButton {...props}>
```

Reply:
Ah, was trying not to pass through all props in case there was overlap with Menu or MenuTrigger
I suppose in cases like that we'd create a specific object for passing through directly to one of those though, so it'd be fine